x-pack/plugins/security_solution/server/search_strategy/endpoint_package_policies_stats/index.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ISearchStrategy, SearchStrategyDependencies } from '@kbn/data-plugin/server';
+
+import { from } from 'rxjs';
+import moment from 'moment';
+import type { IKibanaSearchResponse } from '@kbn/data-plugin/common';
+import type { PackagePolicy } from '@kbn/fleet-plugin/common';
+import type { EndpointAppContextService } from '../../endpoint/endpoint_app_context_services';
+import { EndpointAuthorizationError } from '../../endpoint/errors';
+
+export const endpointPackagePoliciesStatsSearchStrategyProvider = (
+  context: EndpointAppContextService
+): ISearchStrategy<{}, IKibanaSearchResponse<{ outdatedManifestsCount: number }>> => {
+  return {
+    search: (_, __, deps) => {
+      return from(requestEndpointPackagePoliciesStatsSearch(context, deps));
+    },
+  };
+};
+
+export const requestEndpointPackagePoliciesStatsSearch = async (
+  context: EndpointAppContextService,
+  deps: SearchStrategyDependencies
+) => {
+  const { canReadEndpointList } = await context.getEndpointAuthz(deps.request);
+
+  if (!canReadEndpointList) {
+    throw new EndpointAuthorizationError();
+  }
+
+  const fleetServices = context.getInternalFleetServices();
+  const perPage = 10000;
+  let page = 1;
+  let allItems: PackagePolicy[] = [];
+
+  while (true) {
+    const rawResponse = await fleetServices.packagePolicy.list(deps.savedObjectsClient, {
+      perPage,
+      page,
+      kuery: `ingest-package-policies.package.name:"endpoint"`,
+    });
+    allItems = allItems.concat(rawResponse.items);
+    if (allItems.length >= rawResponse.total) {
+      break;
+    }
+
+    page++;
+  }
+  const outdatedManifestsCount = allItems.reduce((acc, item) => {
+    const endpointInput = item.inputs.find((input) => input.type === 'endpoint');
```

Comment:
Is this really needed? Since you added the `kuery` above to filter by package name `endpoint`, all package policies should be from `endpoint` I guess. Did you see any different behaviour?

Reply:
Wasn't sure about it - if I fetch `package.name: 'endpoint'`, will it result in policies with only one input of type `endpoint` ?

Reply:
Since you are searching for package policies, yes. This is what we do in other areas like manifest manager: https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts#L678-L687

Reply:
I see, ok, removed the check and went with `inputs[0]`